flax/linen/normalization.py

```diff
@@ -293,7 +293,7 @@ class GroupNorm(Module):
       bias_init: Initializer for bias, by default, zero.
       scale_init: Initializer for scale, by default, one.
   """
-  num_groups: int = 32
+  num_groups: int | None = 32
```

Comment:
nit: Optional[None] so it is in line with group_size type hint?

Reply:
That would make both num_groups and group_size optional, which isn't the case right? num_groups can only be None when group_size is not specified.

Reply:
I don't follow. How does the current PR enforce that num_groups or group_size is specified?

Reply:
Yeah you're right. I was confused. Forced push with the Optional[int] style.